scripts/write-config.sh

```diff
@@ -0,0 +1,206 @@
+
+  <vaulttoken> defaults to the token found in ~/.vault-token.
+
+  <outputdir> defaults to "config/", so when run in the gradle rootdir, it will be
+  in the expected place for automation.
+EOF
+ exit 1
+}
+
+# Get the inputs with defaulting
+default_target=${WSM_WRITE_CONFIG:-local}
+target=${1:-$default_target}
+vaulttoken=${2:-$(cat "$HOME"/.vault-token)}
+outputdir=${3:-config}
+
+# The vault paths are irregular, so we map the target into three variables:
+# k8senv    - the kubernetes environment: alpha, staging, or integration
+# namespace - the namespace in the k8s env: alpha, staging, dev, or the target for personal environments
+# fcenv     - the firecloud delegated service account environment: dev, alpha, stagin
+
+case $target in
+    help | ?)
+        usage
+        ;;
+
+    clean)
+        rm $(pwd)/${outputdir}/* # &> /dev/null
+        exit 0
+        ;;
+
+    local)
+        k8senv=integration
+        namespace=dev
+        fcenv=dev
+        ;;
+
+    dev)
+        k8senv=integration
+        namespace=dev
+        fcenv=dev
+        ;;
+
+    alpha)
+        k8senv=alpha
+        namespace=alpha
+        fcenv=alpha
+        ;;
+
+    staging)
+        k8senv=staging
+        namespace=staging
+        fcenv=staging
+        ;;
+
+
+    *) # personal env
+        k8senv=integration
+        namespace=$target
+        fcenv=dev
+        ;;
+esac
+
+# Create the output directory if it doesn't already exist
+mkdir -p ${outputdir}
+
+# If there a config and it matches, don't regenerate
+if [ -e target.txt ]; then
+    oldtarget=$(<target.txt)
+    if [ "$oldtarget" = "$target" ]; then
+        echo "Config for $target already written"
+        exit 0
+    fi
+fi
+
+# Read a vault path into an output file, decoding from base64
+function vaultgetb64 {
+    vaultpath=$1
+    filename=$2
+    docker run --rm -e VAULT_TOKEN=${vaulttoken} broadinstitute/dsde-toolbox:consul-0.20.0 \
+        vault read -format=json ${vaultpath} | jq -r .data.key | base64 -d > ${outputdir}/${filename}
+}
+
+# Read a vault path into an output file
+function vaultget {
+    vaultpath=$1
+    filename=$2
+    docker run --rm -e VAULT_TOKEN=${vaulttoken} broadinstitute/dsde-toolbox:consul-0.20.0 \
+        vault read -format=json ${vaultpath} | jq -r .data > ${outputdir}/${filename}
+}
+
+# Read database data from a vault path into a set of files
+function vaultgetdb {
+    vaultpath=$1
+    fileprefix=$2
+    fil=${outputdir}/dbtmp.json
+    docker run --rm -e VAULT_TOKEN=${vaulttoken} broadinstitute/dsde-toolbox:consul-0.20.0 \
+        vault read -format=json ${vaultpath} | jq -r .data > ${fil}
+    cat ${fil} | jq -r '.db' > ${outputdir}/${fileprefix}-name.txt
+    cat ${fil} | jq -r '.password' > ${outputdir}/${fileprefix}-password.txt
+    cat ${fil} | jq -r '.username' > ${outputdir}/${fileprefix}-username.txt
+    rm ${fil}
+}
+
+vaultget "secret/dsde/firecloud/${fcenv}/common/firecloud-account.json" "user-delegated-sa.json"
```

Comment:
Now we're publishing details about the organization of our vault setup. From an OpSec perspective, maybe this script should live in a private repo?

Reply:
This is not new behavior. If it is an issue, it is an issue across all of the repos.

Reply:
I mean...we're making public details that are of no use to anyone else except for an attacker. That's my threshold for keeping things private. But yes, easier said than done.